### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Generated by https://github.com/jsoref/spelling `f`; to maintain your repo, please consider `fchurn`

Note: this is one of the largest PRs I've made of late. GitHub doesn't approve of PRs w/ more than 250 commits, and this definitely exceeds that. The vast majority of changes are to the Changelog.  If you exclude that, it would be easier to review.

Unfortunately, due to the size of this PR, GitHub's review facilities will not be particularly helpful. I have tagged a couple of commits w/ `[API]` because I suspect they may be objectionable.

I'm happy to split/adjust/drop things in order to make merging easier.

Probably the easiest thing would be to split out the Changelog changes. Note that at least one of the changes to the changelog appears to be damage correction: 1393f9e3b683f592db08dda777452418df8ebaf9

A quick summary of changes made:
```
-abilit
-ablity
-absoulutely
-abstration
-acces (acces, access)
-accidentely
-accidential
-accidently (Accidently, accidently)
-accoding
-accomodate
-accross
-achive
-actin
-actua
-actuall
-actully
-acurate
-acutally (Acutally, acutally)
-adapation
-adaption
-addtional
-Aditional
-adress
-adresses
-adressing
-agressive
-agressively
-alertig
-algorightm
-allmost (Allmost, allmost)
-ambigious
-ammend (ammend, ammended)
-amont
-amout
-analyer
-analyse
-analysing
-anaylsis
-anaylzer
-anaylzing
-annoance
-anonnymize
-ansync
+anonymize
-Appen
-appliability
-appropiate
-approprate
-aproximation
-aquainted
-aquire
-aquired
-aquiring
-arbritrary
-assgined
-associtated
-asusmption
-atomicitiy
+atomicity
-autentication
-authorative
-autogenernated
-automatted
-automic
+autogenerated
-avaiable
-availabe
-availality
-avalability
-baching
-backbrounding
-backgrouding
+backgrounding
-beging
-begining
-beginnig
-beginnning
-begung
-behaviour (BEHAVIOUR, Behaviour, behaviour)
-betwen
-betwwen
-bsbawueprod
-bufgfix
-bufgix
-bufix
-buid
-buidbot
-builbot
-bw
-caling
-caluculate
-CANEL
-canelling
-Cann't
-cannnot
-canoninized
-capablity
-casesensitve
-casesenstive
-caus (caus, caused)
-causded
+casesensitive
-cerfificate
-Certifcate
-charactes
-charakter
-charcters
-checkig
-choosen
-circumsances
-cirumstances
-claches
-classigal
-closly
-clould
-clumpsy
-codition
-cofig
-comand
-commen
-commiting
-comparision (comparision, comparisions)
-compatibile
-compatiblity
-compatility
-complate
-compresion
-conditons
-coneptual
-Conext
+conffile
-configuation
-configurabel
-configuraton
-configurd
-configureable
-confile
-conncection
-connexion
-consequtive
-consitent
-consomer
-constructer
-constructur
-consturcts
-consumme
-containt
-containung
-contibutions
-continous (Continous, continous)
-continously
-contribued
-controling
-contstant
-conunter
-Copntrol
-copyied
-corase
-coses
-Cound
-countes
-couting
-coverity
-coverty (Coverty, coverty)
+coverity (Coverity, coverity)
-curently
-curret
-currrent
+datagram
-datgram
-dealock
-debateble
-debuger
-debuggung
-debuging
-decermenting
-deciced
+decrementing
-dedected
-defintion (defintion, defintions)
-deflat
-defuncnt
-degredation
-delayble
-delemited
-delemiter (delemiter, delemiters)
-delemiting
-DELIBERATLY
+delayable
-delimition
-delmi
-delmiter
-Delte
+delimitation
-dependcy
-Dependecies
-depite
-deplyments
-deprectated
-depricate
-depricated
-deprication
-dept
-dequeueing
-derefernce
-desasterous
+dequeuing
+dereference
-descandants
-descripters
-descriptiors
-Descrption
-DESRIPTORS
-destinatin
-destructer
-destructur
-detailled
-Deteced
-determinded
-differnt
-directiories
-directroy
-discardmrk
-discared
-disply
-distinguised
-documenation
-documentas
-doesn
-drainging
-droping
-dyamically
-dynafils
+dynafile
-econdings
-effcient
-el
-elaboreate
-elasped
-eleasticsearch
-elswhere
-emmit
-emmitted
-emphermeral
-emptyness
-emulaton
+encodings
-endoints
-endpoint
+endpoint (endpoint, endpoints)
-endutend
-enebale
+endutent
-enqueing
-enqueueing
-enrcyption
-entitiy
+enqueuing
-environmet
-envison
-envrionment (envrionment, envrionments)
-erorrs
-erors
-errmesg
-errornous
-errornously
-erros
-esentially
-espcially
-everthing
-evertyhing
-everyting
-evrything
-EWOULDBOLOCK
-excactly
-excercise
-excpect (excpect, excpected)
+EWOULDBLOCK
-exernal
-exisiting
-existance
-existin
-expaned
-experienceing
-experimantal
-experimnentalls
-explicitely
-explicitley
-explict
-explitely
-exprience
-extention
-exterme
-extracton
-extremly
-faciltity
-facitate
-faile (faile, failes)
-failuere
-favour
-Finsh
-fmhhash
-focuspointcify
-follwing
-forground
-forice
-formater
+formatter
-forwardig
-fragement
-freee
-freqency
-frequence
-fromat
-fuction
-functionalit
-functon
-garbagge
-GB
-genenrate
-Gerhardds
-gloablly
-globl
-grabble
-grabled
-gracefull
-grammer
-guranteed
-gurantees
-hande
-hander
-handlign
-handly
-hanlers
-happend (Happend, happend)
-happer
-hardned
-hashstep
+hashstep (Hashstep, hashstep)
-Haststep
-havy
-healt
-heavey
-hl
-hower
-Howerver
-hundered (hundered, hundereds)
-ignoce
-IGNOPRED
-immediatly
-immeditely
-implementaton
-Implementor's
-implemet
-implicitely
-implmented
-imporper
-improvate
-improvded
-improvment (improvment, improvments)
-imtuxedolog
-inalid
-incidently
-includding
-incomple
-incompleted
-incompliant
-incorect
-increas (increas, increased)
-incrementially
+incrementally
-incremeting
-indicatio
-indicte
-individial
-indizes
-ineffcient
-informaton
-ingore (INGORE, ingore)
-inherant
-inidividual
-ininit
-inital
-initalised
-initalized
-initaliziation
-initialiaze
-initialisaion
-initialisation (Initialisation, initialisation)
-initialise
-initialised
-initializiation
-initialze
-initilizer
-initilizes
-injecmsg
+initializer
-inlcude
-inlcuding
-inotfiy
-inplements
-instace
-instanc
-instanciable
-instanciated
-instantance
-instructin
-instructons
+instantiable
+instantiated
-intall
-inteded
-intension
-intensional
-intensionally
-interdependant
-interger
-interpreation
-interupted
-inthe
-intial
-INTIALIZATION
-intialize (Intialize, intialize, intializes)
-intializing
-intiated
-introdcued
-invald
-involed
-Itme
-jlr
-jorunal
-journalctrl
+journalctl
-jportal
-jw
-kafaj
-kafak
-kafaka
-Kortalainen
+Kortelainen
-landesrecht
-lazyness
-leakd
-leftever
-legay
-leightweight
-lengh
-lenght
-lenghty
-Levhshin
+Levshin
-libary
-liblogging
-Libloging
+liblogging (Liblogging, liblogging)
-libmaxmindb
+libmaxminddb
-likelyhood
-Limt
-limter
-linet
-linke
+limiter
-listner (Listner, listner, listners)
-litteral
+Listner
-lodable
-Logreged
-longitud
-Luckly
-magnitued
-maintened
-maitained
-maitaining
-makign
-malicouos
-malicous
-mangeling
-MAXCOMMUNITYLENGHT
+MAXCOMMUNITYLENGTH
-maxlistners
+maxlisteners
-maxmindb
-MAXOIDLENGHT
-MAXPORTLENGHT
+maxminddb
+MAXOIDLENGTH
+MAXPORTLENGTH
-meachanism
-mechanisims
-memeber
-memmory
-memomry
-mempory
-mesage
-messagt
-messe
-messsage (messsage, messsages)
-Messure
-methode
-minumum
-misadress (misadress, misadressed)
-misadresseing
-misadressing
-misbehaviour
+misaddress (misaddress, misaddressed)
+misaddresseing
+misaddressing
-mmkubertnetes
-mmpstructdata
-modifcation
-modifed
-modificaton
-modifiction
-MODIFID
-Monito
-monogodb
-motify (Motify, motify)
+Motify
+Msgseparator
-mssages
+Multifunctional
-mutally
-Mutifunctional
-neccesarily
-neccessarily
-neeed (neeed, neeeded)
-netstrmsworking
-Neverthless
+netstrmworking
+noncompliant
-normalizaton
-notfied
-noticable
-noticible
-notifcation
-notificaton
-nubmer
-numbe
-occurence (occurence, occurences)
-occuring
-ocurs
-omampq
-omaqp
-omcmzq
+omczmq
-omelasticsearc
-omelasticsearch
-omeleastisearch
+omelasticsearch (Omelasticsearch, omelasticsearch)
-Omleasticsearch
-omrabittmq
-omusrsmsg
-operatons
-opertion
-optimisation (optimisation, Optimisations)
-optimizatiions
-optinally (Optinally, optinally)
-optiomization
-oputput
-orginal
-orginally
-orginating
-Orignal
-orignally (Orignally, orignally)
-ouput
-ourselfes
-ourselvs
-outout
-overisze
-overlow
-overriden
-overrruns
-overwirtten
-paramater
-paramter (Paramter, paramter, paramters)
-paranthesis
-parctical
-paremeters
-parseing
-Passig
-passwed
-patche (patche, patches)
-patcjh
-pathes
-performa
-Perists
-permissons
-permites
-persistance
-persistet
-persits (Persits, persits)
-persmissions
-pervious
-ponter
-portlets
-posibility
-poskey
-postion
-postively
-potentiall
-potentielly
-preceeding
-precison
-preferrable
-preferrably
-preferrence
-preperation
-prequisite (prequisite, PREQUISITES, prequisites)
-presense
-preseves
-Previosly
-Previsouly
-priveleges
-priviledge (priviledge, priviledges)
-priviledged
-problemns
-procesing
-processig
-proides
-prooven
-properply
-propert
-propertis
-propety
-propper
-proprety
-propsing
-propterty
-proptery
-protectes
-protype
-proveded
-proviced
-providig
-provinding
-Prozess
-psml
-pstatsinverval
+pstatsinterval
-pthead
-puling
-pyhsical
-Qeue
-quering
-racieness
-raesons
-rainrescript
-rater
-readble
-readibility
-readign
-readybility
-readyble
-realy
-reanming
-receiption
-receveived
-Recieve
-recieved
-recognice
-recognising
-recomendation
-recption
-recration
-redheat
+redhat
-reentrency
+Reenabled
+reentrancy
+refactored
-refectored
-refeence
-regeression
-registerred
-regradless
-reinstanciated
-reliablity
-rellay
+reinstantiated
-relptuil
-remaing
-Remeber
-remporary
-Rempote
-Renabled
-repated
-repetetively
+relputil
+repetitively
-reportig
-reproting
-Requetsed
-requred
-requried
-requries
-resambles
-Resett
-reseve
-reslolves
-reslult
-reslution
-resouces
-ressource (ressource, Ressources, ressources)
-ressoure
-restablish
-resultion
-retreiving
-returend
-revelevant
-rgerhad (rgerhad, rgerhads)
-rgerhards
-rgerhars
+rgerhard (rgerhard, rgerhards)
-rpbuild
-rpmbuld
-rserved
-rsylog's
-rsylsog
-rsyslod
-rsysog
-rsysogd
-rulengine
+ruleengine
-rulset
-runing
-runnnig
-runtuime
-russion
+russian
-sanitazion (Sanitazion, sanitazion)
+sanitization (Sanitization, sanitization)
-sanitzier
-segaults
-semiconlon
-seperate
-seperated
-seperately
-seperatly
-seperator
-serializiation
-seriazable
-servcies
+serializable
-seting (seting, setings)
-settigs
-settingnot
-severits
-shool
-shoud
-shoulod
-showdoccase
-shutdow
-Signatur
-SIGTEM
+SIGTERM
-sigup
-similiar
-simlifies
-simplictic
-simplificytin
-simplyfies
-singel
-SIOCGIFCOMF
-SIOCGLIFCONF
+SIOCGIFCONF
+SIOCGLIFCOMF
-slection
-slighly
-smaple
-smoothens
-soley
-somehting
-somehwat
-somethting
-someting
-sorce
-specfied
-specifc
-specifiy
-speficied
-spehttp
-speicifes
-spliting
-spoofadr
+spoofaddr
-squased
-sring
-statemetn
-statment (statment, statments)
-stdouot
-stipped
-stoping
-stratup
-streamdrivermmode
-STREAMMMODE
-strelen
+streamdrivermode
+STREAMMODE
-stricly
+strlen
-strucuture
-struture
-subleties
-subract
-substing
-subsyste
-subsytem
-Subtree
-succceds
-succes (succes, SUCCESS, success)
-SUCCESFUL
-succesfully
-successfull
-sucessfull
-sucide
-sufficently
-superflous
-superseeded
-suport
-suppored
-supportd
-suppressions
-supress
-supressing
-supression (supression, supressions)
-supressor
+suppressor
-suspeneds
-sylsogd
-symlinkl
-symlinks
+symlink (symlink, symlinks)
-systemn
-systen
-Sytem
-szenario
-taget
-temporay
-teplate
-termiante
-termnator
-themselfes
-themsleves
-theoritically
-ther
-thight
-thistime
-thory
-thourough
-thouroughly
-throtteling
-thuis
-timeing
-timeout (TIMEOUT, Timeout, timeout)
+timeout (TIMEOUT, Timeout, timeout, timeouts)
-timout (timout, timouts)
-timstamp
-toc (TOC, toc)
-togehter
-transation
-transcation
-transistion
-transmited
-treshold
-triggerred
+treshhold
-trival
-Trucate
-trucation
-truely
-trunctates
-trunctation
-tupels
+tuples
-Umw
-unaccessible
-unavailibility
-uncomitted
+unavailability
-uncommited
-underlaying
-undfined
-unecessary
-uner
-unexcpected
-unexepectedly
-unfished
-Unfortuantely
-Unfortunaltely
-unfortunatly
-Unforutnately
-unitialized
-unitinialized
-UNKNWON
-unkonwn
-unkown
-unlikley
-unlikly
-unnoticable
-unstrectured
-unsuable
-unsuccesfull
-unupported
-Unuse
-unvailable
-unvailablity
-USACSII
-usally
+USASCII
-utilties
-valule
-varmjofekoj
-varmofekoj
-varmojefkoj
-varmojfeko
-varmojkekoj
-verifcation
-verifiy
-versin
-veryfing
-visiable
-voluntarly
-vrebose
-Vw
-warings
-warnigs
-warnomg
-wating
-weired
-whcih
-whish
-whith
-whitspace
-wich
+whitespace
-Windo
-wirting
-witin
-workaroud
-Wrahmen
-writting
-wvait
-zookeper
```

(Quickly scanning this list doesn't show anything terribly wrong, but that doesn't guarantee that my corrections are correct, it should mean that I haven't made any egregious errors while applying my replacements.)

### CODE REVIEW

**Comment by jsoref** [comment]
I don't eagerly squash as it makes it very hard to drop changes in cases when that is necessary.

Once people are happy w/ the content, I'm happy to squash commits.

**Comment by davidelang** [comment]
On Sun, 22 Dec 2019, Josh Soref wrote:

> I don't eagerly squash as it makes it very hard to drop changes in cases when that is necessary.
>
> Once people are happy w/ the content, I'm happy to squash commits.

I haven't looked through this yet, but where there are a lot of changes like 
this, please use the interactive rebase option (and force push as needed) to 
adjust the commits to look like you were perfect.

i.e. don't have commit 15 fix commit 5, re-order the commits to group them 
together and then squash the two into one so that it looks like you never made a 
mistake on commit 5

you don't want to edit commits like this after it's merged, but while it's 
pending merge, (and while you are tweaking it yourself) it should be fair game.

David Lang

**Comment by jsoref** [comment]
@davidelang: thanks, yes, I'm currently force-pushing rewritten "perfect" commits

**Comment by jsoref** [comment]
@davidelang: Not that I'd include a fix for this in this PR (it's out of scope), but would a PR that addressed the complaint in https://www.codefactor.io/repository/github/rsyslog/rsyslog/pull/4063 -- i.e. switching from
```bash
foo;
if [ ! $? -eq 0 ]; then
```
to
```bash
if ! foo; then
```
be accepted?

**Comment by rgerhards** [comment]
Thx for the PR, much appreciated. I will squash the commits once the PR is ready for merge.



> Not that I'd include a fix for this in this PR (it's out of scope), but would a PR that addressed the complaint in https://www.codefactor.io/repository/github/rsyslog/rsyslog/pull/4063 -- i.e. switching from

I am skeptic on doing this automatically - sometimes the resulting new constructs are very hard to read IMHO.

**Comment by jsoref** [comment]
I'll see about splitting out the ones that appear to be API-ish (I've split one commit out because it was tied to https://github.com/rsyslog/librelp).

Possibly some evening this week, possibly next weekend/week.

The CI is down to two failures and I haven't been able to determine their corresponding changes.

**Comment by jsoref** [comment]
Fwiw, virtually nothing I do is "automatic". Although, for the ~14/26 issues CodeFactor is complaining about, I suspect I'd be more error-prone than someone else doing it. As such, I'm probably not going to.

**Comment by jsoref** [comment]
@rgerhards: I really can't figure out what's special about the remaining failures.

File: Makefile.am [hunk 1]
```diff
@@ -535,7 +535,7 @@ DISTCHECK_CONFIGURE_FLAGS+= --enable-mmtaghostname
 endif
 
 if ENABLE_IMTUXEDOULOG
-DISTCHECK_CONFIGURE_FLAGS+= --enable-imtuxedolog
+DISTCHECK_CONFIGURE_FLAGS+= --enable-imtuxedoulog
```
Comment (rgerhards) [inline Makefile.am]: This cannot be changed, it breaks backward copatibility - plus it makes little sense, as this is the name of a component.

File: action.c [hunk 2]
```diff
@@ -304,7 +304,7 @@ actionResetQueueParams(void)
 	cs.iActionQueueNumWorkers = 1;			/* number of worker threads for the mm queue above */
 	cs.iActionQueMaxFileSize = 1024*1024;
 	cs.iActionQPersistUpdCnt = 0;			/* persist queue info every n updates */
-	cs.bActionQSyncQeueFiles = 0;
+	cs.bActionQSyncQueueFiles = 0;
```
Comment (rgerhards) [inline action.c]: this also potentially breaks backward compatibility

File: Makefile.am (hunk 1 above)
Comment (jsoref) [reply Makefile.am]: Sorry, the correct fix is: https://github.com/jsoref/rsyslog/commit/6bc5a94d64a0a10248a13156a1e354421a44b787

What I'm doing is looking at two tokens and picking one, I picked the wrong one.

File: contrib/mmgrok/mmgrok.c [hunk 3]
```diff
@@ -284,33 +284,33 @@ parse_result_store(const grok_match_t gm,instanceData *pData)
 	RETiRet;
 }
 
-/* motify message for per line */
+/* notify message for per line */
```
Comment (jsoref) [inline contrib/mmgrok/mmgrok.c]: Thinking about this a bit more, I suspect it's probably `modify` not `notify`

File: plugins/imdiag/imdiag.c [hunk 4]
```diff
@@ -264,18 +264,18 @@ injectMsg(uchar *pszCmd, tcps_sess_t *pSess)
...
 	DEFiRet;
 
-	litteralMsg = NULL;
+	literalMsg = NULL;
 
 	memset(wordBuf, 0, sizeof(wordBuf));
 	CHKiRet(ratelimitNew(&ratelimit, "imdiag", "injectmsg"));
 	/* we do not check errors here! */
 	getFirstWord(&pszCmd, wordBuf, sizeof(wordBuf), TO_LOWERCASE);
-	if (ustrcmp(UCHAR_CONSTANT("litteral"), wordBuf) == 0) {
```
Comment (jsoref) [inline plugins/imdiag/imdiag.c]: I can't tell if this is an external API or a private thing for testing.

**Comment by jsoref** [comment]
@rgerhards: ok, I think this should be reviewable. The remaining failures are:
* CodeFactor which we've discussed https://github.com/rsyslog/rsyslog/pull/4063#issuecomment-568396062
* PR structure validation which you said you'd address https://github.com/rsyslog/rsyslog/pull/4063#issuecomment-568396062

File: contrib/mmgrok/mmgrok.c (hunk 3 above)
Comment (jsoref) [reply contrib/mmgrok/mmgrok.c]: I've moved this to https://github.com/rsyslog/rsyslog/commit/ebca5bc0327c53375ff8b833aefeac1d1674d06c

File: plugins/imdiag/imdiag.c (hunk 4 above)
Comment (jsoref) [reply plugins/imdiag/imdiag.c]: I've moved this to https://github.com/rsyslog/rsyslog/commit/e6764a021ca6573f71dbcdfd9d45db229676c874

**Comment by jsoref** [comment]
The remaining changes now live here: https://github.com/jsoref/rsyslog/commits/spelling-api
Note that I'm aware some of those commits should be dropped as they're just wrong. I'll revisit that branch after this PR is complete..

**Comment by rgerhards** [comment]
I intend to merge this in three steps:

1. pure text files - no risk at all
2. testbench - very low to now risk
3. code files - need to review them in depth

Would you like to propose the three PRs or should I work out of this PR here? For me it really doesn't matter, but I wanted to ask :-)

**Comment by rgerhards** [comment]
@jsoref oh, one thing: I guess you touched some files which are still GPL. We are moving towards ASL 2.0. Do you agree to release all of your changes under ASL 2.0?

**Comment by jsoref** [comment]
I'm happy to release under both.

I'm also happy to let you do the split from here. Thanks

**Comment by rgerhards** [comment]
thx, I'll proceed as needed :-)

**Comment by rgerhards** [comment]
This introduces too many problems, merged what was unproblematic.

**Pull request closed without merging**